mbuild/compound.py

```diff
@@ -1154,7 +1155,248 @@ def _kick(self):
             particle.pos += (np.random.rand(3,) - 0.5) / 100
         self._update_port_locations(xyz_init)
 
-    def energy_minimization(self, steps=2500, algorithm='cg',
+    warning_message = 'Please use Compound.energy_minimize()'
+    @deprecated(warning_message)
+    def energy_minimization(self, forcefield='UFF', steps=1000, **kwargs):
+        self.energy_minimize(forcefield=forcefield, steps=steps, **kwargs)
+
+    def energy_minimize(self, forcefield='UFF',steps=1000, **kwargs):
+        """Perform an energy minimization on a Compound
+
+        Utilizes Open Babel (http://openbabel.org/docs/dev/) to perform an
+        energy minimization/geometry optimization on a Compound by applying
+        a generic force field
+
+        Can also utilize OpenMM (http://openmm.org/) to energy minimize
+        after atomtyping a Compound using
+        Foyer (https://github.com/mosdef-hub/foyer) to apply a forcefield
+        XML file that contains valid SMARTS strings.
+
+        This function is primarily intended to be used on smaller components,
+        with sizes on the order of 10's to 100's of particles, as the energy
+        minimization scales poorly with the number of particles.
+
+        Parameters
+        ----------
+        steps : int, optional, default=1000
+            The number of optimization iterations
+        forcefield : str, optional, default='UFF'
+            The generic force field to apply to the Compound for minimization.
+            Valid options are 'MMFF94', 'MMFF94s', ''UFF', 'GAFF', and 'Ghemical'.
+            Please refer to the Open Babel documentation (http://open-babel.
+            readthedocs.io/en/latest/Forcefields/Overview.html) when considering
+            your choice of force field.
+            Utilizing OpenMM for energy minimization requires a forcefield
+            XML file with valid SMARTS strings. Please refer to (http://docs.
+            openmm.org/7.0.0/userguide/application.html#creating-force-fields)
+            for more information.
+
+
+        Keyword Args
+        ------------
+        scale_bonds : float, optional, default=1
+            Scales the bond force constant (1 is completely on).
+            For _energy_minimize_openmm
+        scale_angles : float, optional, default=1
+            Scales the angle force constant (1 is completely on)
+            For _energy_minimize_openmm
+        scale_torsions : float, optional, default=0.5
+            Scales the torsional force constants (1 is completely on)
+            For _energy_minimize_openmm
+        scale_nonbonded : float, optional, default=0.75
+            Scales epsilon (1 is completely on)
+            For _energy_minimize_openmm
+        algorithm : str, optional, default='cg'
```

Comment:
if we default to openbabel, should put this kwarg at the top of this list of definitions. Just to make it easier to find the default behavior.